### ISSUE TITLE
[HOLD on #9382][$2000] Main chat is slow to load (15-30 seconds) on desktop

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#8915]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Sent message in Desktop to Hax
2. Closed lid of computer
3. Hax sends message to me
4. Saw incoming message from Hax on phone
5. Opened laptop and NewDot

## Expected Result:
To see new chats from users

## Actual Result:
No new display chats for a while.. 30 seconds

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Desktop App

**Version Number:** 1.1.57-7
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
[slow load desktop may 9 copy.txt](https://github.com/Expensify/App/files/8657266/slow.load.desktop.may.9.copy.txt)

**Upwork URL: https://www.upwork.com/jobs/~01fcac73e3ce9fa2d7
**Issue reported by:** @mallenexpensify 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1652103348179719

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by iwiznia** [comment]
Sorry, but I don't understand the steps. Step 3 says you see an incoming message from Hax, but there's no step where Hax sends you a message

**Comment by mallenexpensify** [comment]
Added
> 3. Hax sends message to me

I thought `Saw incoming message from Hax on phone` inferred that Hax actually sent a message :trollface:

**Comment by iwiznia** [comment]
We should investigate why there was a 30 seconds delay. is it because one all was slow? Is it because we don't re-trigger the download logic for a while? Is it that we triggered so many requests that the browser staggered them?

**Comment by mallenexpensify** [comment]
I can try to reproduce if/when needed.  Hopefully the logs have some useful info in 'em

**Comment by trjExpensify** [comment]
Oh interesting, I think @cead22 is working on improving the performance here. Is that right, Carlos?

**Comment by cead22** [comment]
Yup, but it's helpful if we can have contributors look into this to try and answer @iwiznia 's questions above

**Comment by trjExpensify** [comment]
Perfecto. Upwork job is here: https://www.upwork.com/jobs/~01e6d82585cdf1ad72

**Comment by jrstanley** [comment]
Proposal (WIP):

I don't believe it is any of the factors below.

From the logs a Pusher 1006 error can be identified. This is the WebSocket connection closing abruptly as (presumably) the laptop is put to sleep/resumed. Pusher the attempts to reauthenticate and subscribe to the appropriate channels. Only once this process is complete do new messages appear.

Reauthenticating to Pusher can be seen in the logs to take a couple of attempts. These attempts are throttled using loadash, to avoid multiple concurrent Pusher connections, however the wait time would only account for 5s of the approximate 30s.

There is also a mechanism that checks for the connectivity status of the app changing. When reconnecting ("coming back online") this triggers a number of events, including the reauthentication to Pusher. It can be seen from the logs that this change in connectivity state is not always identified. This may be due to the execution of code being halted before the offline status is stored. My next steps will be to determine what, if any, impact the connectivity state changes (or lack of) has on the time taken to reconnect to Pusher.



> We should investigate why there was a 30 seconds delay. is it because one all was slow? Is it because we don't re-trigger the download logic for a while? Is it that we triggered so many requests that the browser staggered them?

**Comment by trjExpensify** [comment]
Not overdue, Melvin. Proposals being discussed!

**Comment by trjExpensify** [comment]
👋 Hey @jrstanley! Are you still interested in this one?

**Comment by trjExpensify** [comment]
Okay, got it. Thanks for the added context. Feel free to use #expensify-open-source in Slack to bounce around some ideas from the wider community, if needed!

**Comment by trjExpensify** [comment]
👋 @jrstanley, did you bring this one up in #expensify-open-source or have any other ideas here?

**Comment by trjExpensify** [comment]
Seems to have gone quiet here. I've [increased the price for the job](https://www.upwork.com/jobs/~01fcac73e3ce9fa2d7) on Upwork.

**Comment by trjExpensify** [comment]
Job doubled on Upwork [here](https://www.upwork.com/jobs/~01fcac73e3ce9fa2d7).

**Comment by flodnv** [comment]
@jrstanley I think if we can fix 7s of delay that would already be a good improvement! Would you be interested in updating your proposal for that as per @parasharrajat's recommendations above?

**Comment by trjExpensify** [comment]
Job doubled on Upwork [here](https://www.upwork.com/jobs/~01fcac73e3ce9fa2d7).

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third  week). @flodnv @trjExpensify Are we ok to close this?

**Comment by flodnv** [comment]
Hmmm... @mallenexpensify can you try again?

**Comment by mallenexpensify** [comment]
Longshot but.. any chance this might be related to https://github.com/Expensify/App/issues/9382#issuecomment-1222187430
And, if so, should be put this on hold since that issue is actively being worked on? 
I just tried to reproduce but the chats didn't load at all (like #9382).  One potential reason for a 15-30 second delay is that a chat could be hidden and not show up then, if a new message is sent once the computer is reopened, the chat shows ¯\_(ツ)_/¯

**Comment by mvtglobally** [comment]
We will keep trying as well 
Issue not reproducible during KI retests. (Fourth week)

**Comment by mallenexpensify** [comment]
Removed `Help Wanted` and added `[HOLD on #9382] to title. 🤞  that helps

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Fifth week). Still trying

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (6th week)

**Comment by trjExpensify** [comment]
6th week and not reproducible. @mallenexpensify can we close?

**Comment by mallenexpensify** [comment]
Yeah, I'll close.  There are still issues with chats not properly showing but I think the current main one has to do with unread chats not being bold in LHN and that should be fixed very soon with https://github.com/Expensify/Expensify/issues/227410.
